crates/router/src/core/payments/operations/payment_complete_authorize.rs

```diff
@@ -100,9 +100,28 @@ impl<F: Send + Clone> GetTracker<F, PaymentData<F>, api::PaymentsRequest> for Co
         let token = token.or_else(|| payment_attempt.payment_token.clone());
 
         helpers::validate_pm_or_token_given(
-            &request.payment_method,
-            &request.payment_method_data,
-            &request.payment_method_type,
+            &request
+                .payment_method
+                .or(Some(api_models::enums::PaymentMethod::foreign_from(
+                    payment_attempt
+                        .payment_method
+                        .ok_or(errors::ApiErrorResponse::PaymentMethodNotFound)?,
+                ))),
+            &request.payment_method_data.clone().or(Some(
+                payment_attempt
+                    .payment_method_data
+                    .clone()
+                    .parse_value("payment method")
+                    .change_context(errors::ApiErrorResponse::PaymentMethodNotFound)?,
+            )),
+            &request.payment_method_type.or(Some(
+                api_models::enums::PaymentMethodType::foreign_from(
+                    payment_attempt
+                        .payment_method_type
+                        .clone()
+                        .ok_or(errors::ApiErrorResponse::PaymentMethodTypeNotFound)?,
```

Comment:
Use the approach recommended above. Also can you move this logic to before calling the function, it eases readability.